### DISCUSSION TITLE
Parses self-closing tags differently than the "old way"?

### DISCUSSION THREAD

**Discussion by dmatsumoto** [zeux/pugixml#513] [Q&A]
I'm using the following XML string for some pugixml unit tests:

```
string buffer =
            "<?xml version='1.0' encoding='ASCII' ?>\n"
            "<Parent file='test' md5sum='8e5db7acffe6249ba3351f50c7e9eb5f' version='1.0' >\n"
            "\t<Child1 child1_attribute='child1_attribute_value' />\n"
            "</Parent>";
```

When I load this buffer with the following code:

```
	pugi::xml_document doc;
	doc.load_string(buffer.c_str());
```

The resulting DOM object has the right data.  *However*, if I use the old style tag, or if I add text to the Child1 element, e.g.:

`"\t<Child1 child1_attribute='child1_attribute_value' ></Child1>\n"`
or
`"\t<Child1 child1_attribute='child1_attribute_value' >child1_text</Child1>\n"`

the DOM object is incorrect.  What happens is that the Child1 node magically gets another child node with a null name, and has a value of "Child1".  I've tried to force my code to ignore this extra node by comparing the type to node_null, but that doesn't appear to work.

Since I'm new to pugixml, I figure I could be doing something wrong -- however, I don't think that's the case, because of the difference in behavior between the self-closing tag and non-self-closing tag.

**Comment by zeux** [comment]
In your second example, you should expect that Child1 node has a child with type node_pcdata and value child1_text. See https://pugixml.org/docs/manual.html#node_pcdata.

In your first example, there's no text between the opening and closing tag so there should be no children - the tree representation should be exactly the same between `<child1/>` and `<child1></child1>`.

**Reply by dmatsumoto** [reply]
@zeux thank you, I clearly have something to learn about proper XML.  My knowledge over the years is probably rudimentary at best.

However, I do see that `<child1 />` and `<child1></child1>` give me different trees.  How can I package this into something that I can show to you?

**Reply by zeux** [reply]
If you can create a standalone example (eg a single `.cpp` file that contains the code to parse and display the tree) I can take a look.

**Reply by dmatsumoto** [reply]
Ok, I'll try to work on that before I leave town!  Thank you.

**Reply by dmatsumoto** [reply]
@zeux Naturally, the test app I've written works just fine.  It's not exactly like the code I was testing earlier, so I'll have to look over it more closely to see what I could have missed.  Or I'll see if I can make the test app closer to my other test, but unfortunately it's mixed in with code that I'm not able to share.
